src/NetTopologySuite/IO/WKTReader.cs

```diff
@@ -61,9 +74,8 @@ public class WKTReader
         /// <param name="geometryFactory">The factory used to create <c>Geometry</c>s.</param>
         public WKTReader(GeometryFactory geometryFactory)
         {
-            _coordinateSequencefactory = geometryFactory.CoordinateSequenceFactory;
-            _precisionModel = geometryFactory.PrecisionModel;
-            DefaultSRID = geometryFactory.SRID;
+            _ntsGeometryServices = NtsGeometryServices.Instance;
+            Factory = geometryFactory;
```

Comment:
this can cause needless confusion and rework when someone passes in a `GeometryFactory` that they created with a custom `NtsGeometryServices` instance that they use consistently without ever bothering to set it as `NtsGeometryServices.Instance`.

it's also confusing to have both constructors, because we don't give any real direction to show how to use it... **minimally**, I think, we should mark this `GeometryFactory` constructor obsolete if we're going to have both, but we can do better to make the upgrade smoother for people coming from earlier versions.

suggestion (most consistent with how this class has always operated):
- remove the new `NtsGeometryServices` constructor
- add another overload to `NtsGeometryServices.CreateGeometryFactory` that lets us pass in a `GeometryOverlay` instance (and all the baggage that goes with it)
- instead of adding the `_ntsGeometryServices` field and the new constructor, let's add a `_geometryOverlay` field like the others and extract it from the `GeometryFactory` / pass it through in the `Factory` property getter to `NtsGeometryServices.Instance.CreateGeometryFactory` like we currently pass through the others
- `Factory` property setter can now extract the `GeometryOverlay` into the field instead of throwing if it's different.

another way to do this (most consistent with how I would develop it in the first place):
- stop extracting all the individual pieces from the `GeometryFactory` or `NtsGeometryServices` instance that we were passed in (`CoordinateSequenceFactory`, `PrecisionModel`, `SRID`).
- instead, fetch all of those and the `GeometryOverlay` instance, when needed, from the `_ntsGeometryServices` field
- in the `GeometryFactory` constructor and `Factory` property setter, create a new `NtsGeometryServices` instance for the `_ntsGeometryServices` field by extracting **all** of the pieces.
    - this would require `_ntsGeometryServices` to no longer be `readonly`.
    - `Factory` property setter can lose the `GeometryOverlay` check now.
- mark the `GeometryFactory` constructor (and the `Factory` property, I guess) `[Obsolete]`, directing callers to use the `NtsGeometryServices` constructor instead (we did this part with `WKBReader` too, it seems)

combining what I feel are the good parts of both (this is what I would personally aim to do here):
- stop extracting all the individual pieces from the `GeometryFactory` or `NtsGeometryServices` instance that we were passed in (`CoordinateSequenceFactory`, `PrecisionModel`, `SRID`) 
- instead, fetch all of those and the `GeometryOverlay` instance when needed, from the `_ntsGeometryServices` field
- in the `GeometryFactory` constructor and `Factory` property setter, create a new `NtsGeometryServices` instance for the `_ntsGeometryServices` field by extracting **all** of the pieces.  this would require `_ntsGeometryServices` to no longer be `readonly`.
- mark the `GeometryFactory` constructor and the `Factory` property `[Obsolete]`, directing callers to use the `NtsGeometryServices` constructor instead (we did this part with `WKBReader` too, it seems)
- add another overload to `NtsGeometryServices.CreateGeometryFactory` that lets us pass in a `GeometryOverlay` instance (and all the baggage that goes with it)
- in the `GeometryFactory` getter, change `NtsGeometryServices.Instance.CreateGeometryFactory(stuff)` to `_ntsGeometryServices.CreateGeometryFactory()`

Reply:
To clarify: one example reason why someone might be setting `NtsGeometryServices.Instance` differently from what they're using elsewhere would be if they're an ASP.NET Core Web API application that gives its downstream users the same choice that we give our own downstream users: "do you want to opt-in to trying the new stuff for this request, knowing that it might change outputs from earlier versions of this application?"

Reply:
I like the combination.